### ISSUE TITLE
[bug] XLM-RoBERTa reranker models fail to load: classifier.weight KeyError + missing position_id offset

### ISSUE THREAD

**Issue by inmanbobby71** [waybarrios/vllm-mlx#545]
## Summary

XLM-RoBERTa-based reranker models cannot be loaded via `--rerank-model` in vllm-mlx 0.3.0. Affects every modern strong reranker I'm aware of: `BAAI/bge-reranker-v2-m3`, `jinaai/jina-reranker-v2-base-multilingual`, the Qwen3-Reranker family — all use the `XLMRobertaForSequenceClassification` architecture. Two distinct bugs in `rerank_forward.py` combine to make these models unusable: one is fatal at first request, the other is a silent correctness bug that surfaces only after the first is patched.

Companion to #542 (registry-mode `_model_name=None`). I hit this while attempting to swap `cross-encoder/ms-marco-MiniLM-L6-v2` → `BAAI/bge-reranker-v2-m3` on an M4 Max daemon and rolled back the same evening.

## Bug 1 — Two-layer classification head not handled (FATAL)

`XLMRobertaClassificationHead` is a `Dense → tanh → out_proj` pattern, so the safetensors weights expose `classifier.dense.weight`, `classifier.dense.bias`, `classifier.out_proj.weight`, `classifier.out_proj.bias`. vllm-mlx's `classifier_forward()` (around `rerank_forward.py:83`) reads a flat `classifier.weight` / `classifier.bias` — the BERT-style single-layer pattern from `ms-marco-MiniLM`.

**Symptom:** first `POST /v1/rerank` after daemon start returns:

```json
{"detail": "'classifier.weight'"}
```

…and the rerank path is dead until the model is swapped back.

## Bug 2 — RoBERTa-family position ID offset missing (SILENT CORRECTNESS)

RoBERTa-family models require `position_ids = arange(padding_idx + 1, seq_len + padding_idx + 1)` with `padding_idx = 1` (see HF Transformers `create_position_ids_from_input_ids`). vllm-mlx uses plain `arange(seq_len)` directly. Once Bug 1 is patched, the model loads and serves requests — but the position embeddings are off-by-(padding_idx+1) and the resulting scores are garbage. No exception is raised; the rerank scores just stop being meaningful. This is the worse of the two bugs because nothing tells you it's broken.

## Reproduction

On a working vllm-mlx 0.3.0 daemon (registry mode or single-model mode, same result):

1. Edit the LaunchAgent plist (or CLI invocation) to add `--rerank-model BAAI/bge-reranker-v2-m3`. Model files (~2.3 GB) pre-downloaded to `HF_HOME`.
2. Restart the daemon. Startup completes cleanly; `/v1/models` lists the model.
3. `curl -s -X POST http://<host>:8000/v1/rerank -H 'Content-Type: application/json' -d '{"model":"BAAI/bge-reranker-v2-m3","query":"q","documents":["d1","d2"]}'`
4. Response: `{"detail":"'classifier.weight'"}`.

Same shape with `jinaai/jina-reranker-v2-base-multilingual` and (per HF model card inspection) the Qwen3-Reranker family. Patching Bug 1 alone moves you to silent Bug 2 territory — at that point an eyeball check against a known-good cross-encoder is the only way to tell scores have gone wrong.

## Suggested patch shape

Untested but matches HF Transformers behaviour. Pasted from my internal post-mortem doc so the maintainer has a starting point.

```python
# In rerank_forward.py classifier_forward():
if "classifier.dense.weight" in weights:
    # Two-layer XLMRobertaClassificationHead: dense -> tanh -> out_proj
    h = mx.tanh(pooled @ weights["classifier.dense.weight"].T + weights["classifier.dense.bias"])
    logits = h @ weights["classifier.out_proj.weight"].T + weights["classifier.out_proj.bias"]
else:  # BERT-style flat head (e.g. ms-marco-MiniLM)
    logits = pooled @ weights["classifier.weight"].T + weights["classifier.bias"]

# In the embeddings section (wherever position_ids is built):
if config.get("model_type") in ("roberta", "xlm-roberta"):
    padding_idx = config.get("pad_token_id", 1)
    position_ids = mx.arange(padding_idx + 1, seq_len + padding_idx + 1)[None, :]
else:
    position_ids = mx.arange(seq_len)[None, :]
```

A `model_type` switch (with fallback to current behaviour) keeps existing BERT-style rerankers working while unlocking XLM-RoBERTa.

## Environment

- vllm-mlx 0.3.0 (from `requirements_v6.lock`, installed via pip from waybarrios/vllm-mlx)
- Apple Silicon: M4 Max 128 GB (also affects M2 Max per family-architecture reasoning, but I haven't repro'd there)
- macOS 15.x (Sequoia)
- Python 3.11 venv, mlx-lm 0.31.3 transitive
- Failing models: `BAAI/bge-reranker-v2-m3`, `jinaai/jina-reranker-v2-base-multilingual` (per architecture; not personally repro'd on Jina)
- Working baseline (BERT-flat head): `cross-encoder/ms-marco-MiniLM-L6-v2` — restored after rollback, continues to serve normally.

## Impact

The set of strong, current-gen rerankers is essentially all XLM-RoBERTa-based right now. Until this lands, vllm-mlx users are pinned to `ms-marco-MiniLM-L6-v2` (good 2019-era 22M-param model) or `cross-encoder/ms-marco-MiniLM-L-12-v2` (50% more params, marginal gain). Anything in the BGE / Jina v2 / Qwen3-Reranker tier is off-limits.

Happy to test a patch on M4 Max + report back. Thanks for the project.

**Comment by Thump604** [comment]
I opened #550 for this.

Scope:
- adds XLM-RoBERTa/RoBERTa two-layer classifier-head support in `rerank_forward.py`
- uses RoBERTa-family position IDs with the padding offset
- preserves existing BERT-style flat classifier heads and BERT position IDs

Validation on the PR:
- red/green tests for the two reported failure modes
- `tests/test_rerank.py`: `41 passed`
- ruff, black check, `git diff --check`
- Runtime upstream-local-repro preflight against clean upstream main

What I am not claiming: I did not run a live BGE/Jina/Qwen3-Reranker quality benchmark locally; this patch covers the code-path failures reported here.
